src/panel/direction/DirectionPanel.jsx

```diff
@@ -206,13 +259,26 @@ export default class DirectionPanel extends React.Component {
     this.setState(previousState => ({
       origin: previousState.destination,
       destination: previousState.origin,
+      originInputText: this.state.destinationInputText,
+      destinationInputText: this.state.originInputText,
       isDirty: true,
     }), this.update);
   }
 
-  changeDirectionPoint = (which, value) => {
-    persistentPointState[which] = value;
-    this.setState({ [which]: value, isDirty: true }, this.update);
+  changeDirectionPoint = (which, value, point) => {
+    persistentPointState[which] = point;
+    this.setState({
+      [which]: point,
+      isDirty: true,
+      [which + 'InputText']: value || '',
+    }, this.update);
+
+    // Retrieve addresses
+    if (which === 'origin') {
+      this.getAddresses(persistentPointState.origin, null);
```

Comment:
I still don't get the point of the `getAddresses` function taking the two points. Among other things, it forces you to make these calls with `null` which look unnatural. As the action on both points is exactly the same, why not having a simpler function?

Reply:
I agreen but if I make this function perform only one request, I don't know how to call it twice without making the two function calls (and the two API calls) synchronous...

Reply:
That's my point, I don't see why you need to synchronize the calls. Is there a UX reason?
Because otherwise, it's like remote work: the less synchronous stuff the better :smile: